### PR TITLE
[rocfft] Successful build in clean chroot

### PR DESCRIPTION
**Pull request by tpkessler**
Reduce dependencies, tidy up build() and package() routines.

### CODE REVIEW

File: rocfft/PKGBUILD [hunk 1]
```diff
@@ -1,32 +1,31 @@
...
 pkgrel=1
-pkgdesc="Next generation FFT implementation for ROCm"
+pkgdesc='Next generation FFT implementation for ROCm'
 arch=('x86_64')
 url="https://github.com/ROCmSoftwarePlatform/rocfft"
 license=('MIT')
-depends=('boost' 'fftw')
-makedepends=("hcc>=$pkgver" 'cmake')
-source=("https://github.com/ROCmSoftwarePlatform/rocFFT/archive/rocm-$_pkgver.tar.gz")
+depends=('hip-hcc')
```
Comment (acxz) [inline rocfft/PKGBUILD:10]: Should this just be `hip` so that `hip-nvcc` and `hip-clang` can work with it?

File: rocfft/PKGBUILD [hunk 2]
```diff
@@ -1,32 +1,31 @@
...
+makedepends=('hcc' 'cmake')
+source=("https://github.com/ROCmSoftwarePlatform/rocFFT/archive/rocm-$pkgver.tar.gz")
```
Comment (acxz) [inline rocfft/PKGBUILD:12]: Can we add ${pkgname}-${pkgver} as shown here: https://wiki.archlinux.org/index.php/PKGBUILD#source

File: rocfft/PKGBUILD [hunk 3]
```diff
@@ -1,32 +1,31 @@
...
 # Maintainer: Markus Näther <naetherm@cs.uni-freiburg.de>
 pkgname=rocfft
-_pkgver=3.3.0
-pkgver="$_pkgver"
+pkgver=3.3.0
 pkgrel=1
-pkgdesc="Next generation FFT implementation for ROCm"
+pkgdesc='Next generation FFT implementation for ROCm'
 arch=('x86_64')
 url="https://github.com/ROCmSoftwarePlatform/rocfft"
```
Comment (acxz) [inline rocfft/PKGBUILD:8]: Can we change the url to https://rocm-documentation.readthedocs.io/en/latest/ROCm_Libraries/ROCm_Libraries.html#rocfft?

File: rocfft/PKGBUILD (hunk 1 above)
Comment (tpkessler) [reply rocfft/PKGBUILD:10]: It only works on the ROCm platform. For CUDA there is cuFFT, part of CUDA.

File: rocfft/PKGBUILD (hunk 2 above)
Comment (tpkessler) [reply rocfft/PKGBUILD:12]: Yes, I'll add it.

File: rocfft/PKGBUILD (hunk 3 above)
Comment (tpkessler) [reply rocfft/PKGBUILD:8]: As I wrote before: The official docu is just a collection of the READMEs from the github repos. If we change the url, we should do this for all packages (at once?).

File: rocfft/PKGBUILD (hunk 1 above)
Comment (acxz) [reply rocfft/PKGBUILD:10]: I see kk

File: rocfft/PKGBUILD (hunk 3 above)
Comment (acxz) [reply rocfft/PKGBUILD:8]: Since its not that important lets just tackle the issue as we see it

**Comment by acxz** [comment]
I receive the following error in clean chroot:
```
CMake Error at /usr/share/cmake-3.17/Modules/CMakeFindDependencyMacro.cmake:47 (find_package):
  By not providing "Findamd_comgr.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "amd_comgr", but CMake did not find one.

  Could not find a package configuration file provided by "amd_comgr" with
  any of the following names:

    amd_comgrConfig.cmake
    amd_comgr-config.cmake

  Add the installation prefix of "amd_comgr" to CMAKE_PREFIX_PATH or set
  "amd_comgr_DIR" to a directory containing one of the above files.  If
  "amd_comgr" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  /opt/rocm/hip/lib/cmake/hip/hip-config.cmake:78 (find_dependency)
  CMakeLists.txt:157 (find_package)
```

**Comment by tpkessler** [comment]
Should be fixed now. I accidentally removed the line from the cmake configuration.

File: rocfft/PKGBUILD (hunk 3 above)
Comment (tpkessler) [reply rocfft/PKGBUILD]: I've updated the URL.

**Comment by acxz** [comment]
After your latest commit I got the following error in clean chroot:
```
-- The CXX compiler identification is Clang 10.0.0
-- Check for working CXX compiler: /opt/rocm/hcc/bin/hcc
-- Check for working CXX compiler: /opt/rocm/hcc/bin/hcc - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning (dev) at CMakeLists.txt:30 (set):
  implicitly converting 'BOOLEAN' to 'STRING' type.
This warning is for project developers.  Use -Wno-dev to suppress it.

Found HIP at /opt/rocm/hip
-- Found HIP: /opt/rocm/hip (found version "3.3.20195-") 
-- HIP Compiler: hcc
CMake Warning (dev) at /usr/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:272 (message):
  The package name passed to `find_package_handle_standard_args` (PkgConfig)
  does not match the name of the calling package (hcc).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  /usr/share/cmake-3.17/Modules/FindPkgConfig.cmake:45 (find_package_handle_standard_args)
  /opt/rocm/hcc/lib/cmake/hcc/ImportedTargets.cmake:1 (include)
  /opt/rocm/hcc/lib/cmake/hcc/hcc-config.cmake:54 (include)
  cmake/VerifyCompiler.cmake:33 (find_package)
  CMakeLists.txt:33 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found PkgConfig: /usr/bin/pkg-config (found version "1.6.3") 
CMake Warning (dev) at /usr/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:272 (message):
  The package name passed to `find_package_handle_standard_args` (PkgConfig)
  does not match the name of the calling package (hcc).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  /usr/share/cmake-3.17/Modules/FindPkgConfig.cmake:45 (find_package_handle_standard_args)
  /opt/rocm/hcc/lib/cmake/hcc/ImportedTargets.cmake:1 (include)
  /opt/rocm/hcc/lib/cmake/hcc/hcc-config.cmake:54 (include)
  /usr/share/cmake-3.17/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /opt/rocm/hip/lib/cmake/hip/hip-config.cmake:77 (find_dependency)
  cmake/VerifyCompiler.cmake:37 (find_package)
  CMakeLists.txt:33 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at /usr/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:164 (message):
  Could NOT find Git (missing: GIT_EXECUTABLE)
Call Stack (most recent call first):
  /usr/share/cmake-3.17/Modules/FindPackageHandleStandardArgs.cmake:445 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.17/Modules/FindGit.cmake:96 (find_package_handle_standard_args)
  cmake/Dependencies.cmake:16 (find_package)
  CMakeLists.txt:45 (include)


-- Configuring incomplete, errors occurred!
See also "/build/rocthrust/src/build/CMakeFiles/CMakeOutput.log".
==> ERROR: A failure occurred in build().
    Aborting...
==> ERROR: Build failed, check /var/lib/archbuild/extra-x86_64/acxz/build
```

Can I ask how you are building in clean chroot? You are followng https://wiki.archlinux.org/index.php/DeveloperWiki:Building_in_a_clean_chroot right? I suggest using `extra-86_64-build` and adding the arch4edu repository in that chroot and adding necessary package deps as needed with the `-I /path/to/built/package`

**Pull request merged**